packages/in_app_purchase/lib/src/in_app_purchase/google_play_connection.dart

```diff
@@ -56,11 +57,8 @@ class GooglePlayConnection
   @override
   void buyConsumable(
       {@required PurchaseParam purchaseParam, bool autoConsume = true}) {
-    if (autoConsume == true) {
-      if (_productIDsToConsume == null) {
-        _productIDsToConsume = Set<String>();
-        _productIDsToConsume.add(purchaseParam.productDetails.id);
-      }
+    if (autoConsume) {
+      _productIdsToConsume.add(purchaseParam.productDetails.id);
```

Comment:
My thinking is that the extra memory usage from the empty set is really minimal, and never letting the set be null in the first place prevents any null exceptions crashing the app down the line.

I think if we do want to avoid the empty initialization we could use Dart's null aware operators to clean it up a little.

```dart
    if (autoConsume) {
      _productIdsToConsume ??= Set<String>();
      _productIdsToConsume.add(purchaseParam.productDetails.id);
```

and then later on:

```dart
if (!(purchaseDetails.status == PurchaseStatus.purchased &&
    _productIdsToConsume?.contains(purchaseDetails.productID))) {
    return purchaseDetails;
}
```

Like I said though I think it's worth the tradeoff here to never have to worry about any null crashes in the first place though. This is article is on Java, not Dart, but I think the reasoning still applies: http://www.javapractices.com/topic/TopicAction.do?Id=59. WDYT?

Reply:
SGTM! Empty set really doesn't matter too much.